pkg/bootstrap/platform/gcp.go

```diff
@@ -207,19 +210,52 @@ func parseGCPMetadata() (pid, npid, cluster, location string) {
 	return envPid, envNpid, envCluster, envLocation
 }
 
+func isZone(z string) bool {
+	return zonePattern.MatchString(z)
+}
+
 // Converts a GCP zone into a region.
 func zoneToRegion(z string) (string, error) {
-	// Zones are in the form <region>-<zone_suffix>, so capture everything but the suffix.
-	re := regexp.MustCompile("(.*)-.*")
-	m := re.FindStringSubmatch(z)
-	if len(m) != 2 {
-		return "", fmt.Errorf("unable to extract region from GCP zone: %s", z)
+	if !isZone(z) {
```

Comment:
did this need to be changed or is it just cleaned up?

Reply:
nevermind, I see it was for later part of the code?

Reply:
I was just cleaning it up.